codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -158,47 +151,49 @@ update these file paths accordingly.
 
   /// The entrance point of examining and code signing an engine artifact.
   Future<void> validateAll() async {
-    for (String passwordFilePath in [
-      codesignAppstoreIDFilePath,
-      codesignTeamIDFilePath,
-      appSpecificPasswordFilePath,
-    ]) {
-      await readPassword(passwordFilePath);
-    }
-    if (availablePasswords.containsValue('')) {
-      throw CodesignException('certian passwords are missing. \n'
-          'make sure you have provided <CODESIGN_APPSTORE_ID>, <CODESIGN_TEAM_ID>, and <APP_SPECIFIC_PASSWORD>');
-    }
-    codesignAppstoreId = availablePasswords['CODESIGN_APPSTORE_ID']!;
-    codesignTeamId = availablePasswords['CODESIGN_TEAM_ID']!;
-    appSpecificPassword = availablePasswords['APP_SPECIFIC_PASSWORD']!;
-
-    await processRemoteZip();
-
-    if (dryrun) {
-      log.info('code signing dry run has completed, If you intend to upload the artifacts back to'
-          ' google cloud storage, please use the --dryrun=false flag to run code signing script.');
+    // The value of codesignedFilePath is used in the cleanup process in the finally block.
+    String? codesignedFilePath;
+    try {
```

Comment:
The code will be more readable if you cleanup the temp file as part of the iteration.

Reply:
Right I was thinking about the clean up process. Currently I told the code to "remove files in root directory except the output path". I was considering the different cases that would trigger a remove if clean up is delegated to the visitors during the iteration, but I think there might be corner cases. I think the current way might be less error prone but I might be wrong, what do you think? If a delegated cleanup is more ideal I can submit it in a followup PR since there would probably be many changes.

Reply:
From what I understand the processing of a single file is a hermetic process. E.g. get the path to the file, unzips it to a temp folder, read the metadata file, signs the relevant files, creates a new zip in the output location and removes the temporary files.

Not sure what corner cases are we expecting to get if we delete the files after processing each file but I see multiple benefits, the code will be cleaner, temp files will be removed right after they are not needed anymore(this is even more important for the bot with the limited disk space), reduce the risk of potentially leaving temp files behind.

Reply:
Oh maybe I have misunderstood "as part of iteration", I thought iteration meant the traversal of the file tree. But now I re-read it and I think maybe it meant delegating cleanup from `validateAll()` to `processRemoteZip()`?

Reply:
I retraced the tmp file generation and changed the cleanup process to be two delete statements, both of which like sir Godofredo said removes the unzipped folders. I think these two places should cover all the cleanup processes. Yeah in deed this looks cleaner than it used to be. Thanks a lot for providing the insights!